### DISCUSSION TITLE
iPXE Menu items do not execute

### DISCUSSION THREAD

**Discussion by rweisman** [ipxe/ipxe#866] [Q&A] [answered]
I am trying to convert to using the ipxe menu.  The issue I am seeing is when I select any of the items I get "Could not boot: No such file or directory ...." then it drops to shell.
If I type the exact command from the shell it works.  I am probably doing something stupid.  

Thanks for help.   Below are the contents of my  ipxmenu.ipxe file.

set boot-url http://xx.xx.xx.xx/tftpboot

menu TestMenu for iPXE
item aape102GB_SAN  AAPE10_Sanboot
item aape102GB_Direct AAPE10_Direct_Sanboot
item debug  Drop to shell

choose target && goto $(target)

:aape102GB_SAN
imgfetch ${boot-url}/aape10_2GB_efi_sanboot.ipxe
imgselect "${boot-url}/aape10_2GB_efi_sanboot.ipxe"
boot

:aap102GB_Direct
sanboot -n http://xx.xx.xx.xx/tftpboot/aape10_2GB/ArrowPE10_amd64_2GB.iso

:debug
shell

**Comment by mcb30** [comment]
Just some minor errors in the script, like using `$(target)` instead of the correct `${target}`.  Here's a corrected version that should work:

```
set boot-url http://xx.xx.xx.xx/tftpboot

menu TestMenu for iPXE
item aape102GB_SAN AAPE10_Sanboot
item aape102GB_Direct AAPE10_Direct_Sanboot
item debug Drop to shell

choose target && goto ${target}

:aape102GB_SAN
chain ${boot-url}/aape10_2GB_efi_sanboot.ipxe

:aap102GB_Direct
sanboot -n http://xx.xx.xx.xx/tftpboot/aape10_2GB/ArrowPE10_amd64_2GB.iso

:debug
shell
```

Once you've got it basically working, you might want to put in some error handling.  For example:
```
choose target && goto ${target}
```
will exit the whole script immediately if the user presses Ctrl-C to cancel the menu.  This may or may not be what you want.  You can use the `||` operator to handle errors, e.g. `choose target && goto ${target} || goto error`.  See https://ipxe.org/scripting for more examples.

**Comment by rweisman** [comment, marked as answer]
I found part of the problem  the boot-url was in (), instead of {}.  Still having the problem the aape102GB_Direct entry.

**Reply by rweisman** [reply]
Ok.. dumb me I found the other problem was a typo.  Sorry if I wasted anybodies time.

**Reply by mcb30** [reply]
Glad you got it working! :slightly_smiling_face:
